forc/src/utils/workspace.rs

```diff
@@ -0,0 +1,149 @@
+use std::{
+    env,
+    ffi::OsStr,
+    path::{Component, Path, PathBuf},
+};
+
+pub fn check_workspace_membership(path: &str, name: &str) {}
+
+pub fn root_manifest(manifest_path: Option<&Path>, cwd: &Path) -> Result<PathBuf, anyhow::Error> {
+    if let Some(manifest_path) = manifest_path {
+        let path = cwd.join(manifest_path);
+        // In general, we try to avoid normalizing paths in Cargo,
+        // but in this particular case we need it to fix #3586.
+        let path = normalize_path(&path);
+        if !path.ends_with("Cargo.toml") && !is_embedded(&path) {
```

Comment:
```suggestion
        if !path.ends_with("Cargo.toml") && !is_embedded(&path) {
```
We should be checking for forc.toml not cargo.toml